src/services/review/RepoCheckerService.ts

```diff
@@ -346,24 +418,56 @@ export default class RepoCheckerService {
         .commit("site checker logs")
         .push(),
       (error) => new SiteCheckerError(`${error}`)
-    )
+    ).map(() => errors)
   }
 
-  checkRepo(repo: string): ResultAsync<undefined, SiteCheckerError> {
+  checkRepo(repo: string): ResultAsync<RepoErrorDto, RepoErrorDto> {
     /**
      * To allow for an easy running of the script, we can temporarily clone
      * the repo to the EFS
      */
 
     let repoExists = true
-
-    return this.cloner(repo).andThen((cloned) => {
-      repoExists = cloned
-      return this.checker(repo)
-        .andThen((errors) => this.reporter(repo, errors))
-        .andThen(() => this.remover(repo, repoExists))
-        .andThen(() => okAsync(undefined))
-    })
+    return this.isCurrentlyLocked(repo)
+      .andThen((isLocked) => {
+        if (isLocked) {
+          return errAsync({
+            status: "loading",
+          })
+        }
+        return this.createLock(repo)
+      })
+      .andThen(() =>
+        this.cloner(repo).andThen((cloned) => {
+          repoExists = cloned
+
+          return this.checker(repo)
+            .andThen((errors) => this.reporter(repo, errors))
+            .map<RepoErrorDto>((errors) => ({
+              status: "success",
+              errors,
+            }))
+
+            .andThen((errors) =>
+              this.remover(repo, repoExists).andThen(() => ok(errors))
+            )
+            .andThen((errors) =>
+              this.deleteLock(repo).andThen(() => ok(errors))
+            )
+        })
+      )
+      .orElse((error) => {
+        if (error.status === "loading") {
+          return okAsync<RepoErrorDto>({
+            status: "loading",
+          })
+        }
+        this.deleteLock(repo)
```

Comment:
in this case, the lock is preserved only if the `status === "loading"` which is fine irght?

Reply:
correct, since some other process is currently running this check!